### ISSUE TITLE
 Error running command "boot2docker,status".: code: 1, msg: error in run: Failed to get machine "Kalabox2": machine does not exist (Did you run `boot2docker init`?)

### ISSUE THREAD

**Issue by dsnopek** [kalabox/kalabox#328]
Everytime I try 'kbox create' I get this error. I tried running 'boot2docker init' and it did it's thing (whatever that is -- I don't have much experience with docker), but 'kbox create' still doesn't work for me. :-/

Any advice?

**Comment by pirog** [comment]
After the npm install did you run kbox provision?

On Thursday, July 23, 2015, David Snopek <notifications@github.com> wrote:

> Everytime I try 'kbox create' I get this error. I tried running
> 'boot2docker init' and it did it's thing (whatever that is -- I don't have
> much experience with docker), but 'kbox create' still doesn't work for me.
> :-/
>
> Any advice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/328>.
>


-- 

Mike

-- 
Mike Pirog
Co-Founder, Products
http://kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by dsnopek** [comment]
Yep! As far as I could tell it was successful, no obvious errors. Does it
make sense to run it again? I can send you the output of it?

2015-07-23 9:02 GMT-05:00 Mike Pirog <notifications@github.com>:

> After the npm install did you run kbox provision?
>

**Comment by pirog** [comment]
Hmmm yeah I would post it. It seems like the VM was not created for you

On Thursday, July 23, 2015, David Snopek <notifications@github.com> wrote:

> Yep! As far as I could tell it was successful, no obvious errors. Does it
> make sense to run it again? I can send you the output of it?
>
> 2015-07-23 9:02 GMT-05:00 Mike Pirog <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>:
>
> > After the npm install did you run kbox provision?
> >
> > On Thursday, July 23, 2015, David Snopek <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> >
> > > Everytime I try 'kbox create' I get this error. I tried running
> > > 'boot2docker init' and it did it's thing (whatever that is -- I don't
> > have
> > > much experience with docker), but 'kbox create' still doesn't work for
> > me.
> > > :-/
> > >
> > > Any advice?
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/kalabox/kalabox/issues/328>.
> > >
> >
> >
> > --
> >
> > Mike
> >
> > --
> > Mike Pirog
> > Co-Founder, Products
> > http://kalamuna.com
> >
> > Stay connected with Kalamuna on: Twitter
> > <
> >
> http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9
> > >
> > , Linkedin
> > <
> >
> http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9
> > >,
> > and Facebook
> > <
> >
> http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/kalabox/kalabox/issues/328#issuecomment-124117812>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/328#issuecomment-124125226>.
>


-- 

Mike

-- 
Mike Pirog
Co-Founder, Products
http://kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by dsnopek** [comment]
Ok, I'm starting over at the top! But I think I might know what the problem is. I ran `npm uninstall kalabox` and decided to try running `npm install kalabox` without the `-g` argument this time, because that installs it as root. I suspect the problem is that kalabox was installed as root, but I wanna use it as my user. I'll let you know how it goes!

**Comment by pirog** [comment]
Ahhh. Yeah this seems to happen a lot. You should install globally but it's
important for the global dirs to be owned by you

On Thursday, July 23, 2015, David Snopek <notifications@github.com> wrote:

> Ok, I'm starting over at the top! But I think I might know what the
> problem is. I ran npm uninstall kalabox and decided to try running npm
> install kalabox without the -g argument this time, because that installs
> it as root. I suspect the problem is that kalabox was installed as root,
> but I wanna use it as my user. I'll let you know how it goes!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/328#issuecomment-124129547>.
>


-- 

Mike

-- 
Mike Pirog
Co-Founder, Products
http://kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by dsnopek** [comment]
When I tried to run everything as my user, I got this error during `kbox provision`:

```
info: -- Step 17/19 --
info: Restarting engine for updates...
info: DOCKER =>  Creating container. Image=kalabox/syncthing:v0.9.0, name=kalabox_syncthing, Binds=undefined, NetworkMode=bridge, 60008/tcp=[HostIp=, HostPort=60008], 22000/tcp=[HostIp=, HostPort=22000], 21025/udp=[HostIp=, HostPort=21025], 21026/udp=[HostIp=, HostPort=21026], VolumesFrom=[kalabox_data], Env=[KBOX_CODEDIR=code, KALABOX=true]
Error running step: {"name":"engine-docker-prepared","description":"Restarting engine for updates...","deps":["core-image-prepare"],"subscribes":[],"isStep":true}: Error while shutting down.: EVENTS =>  Error dispatching event listeners [pre-down].
```

:-/

**Comment by dsnopek** [comment]
Oh, ok, I should install globally, but run as my user and make all the necessary directories be owned by my user? Are there any docs on which directories those are? Or just trial-and-error until works? :-)

**Comment by pirog** [comment]
Yes! In the kalabox read me there are some links to set the correct
permissions right above the install steps I think. Alternatively you can
set your global dir to be in your home folder.

On Thursday, July 23, 2015, David Snopek <notifications@github.com> wrote:

> Oh, ok, I should install globally, but run as my user and make all the
> necessary directories be owned by my user? Are there any docs on which
> directories those are? Or just trial-and-error until works? :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/328#issuecomment-124134242>.
>


-- 

Mike

-- 
Mike Pirog
Co-Founder, Products
http://kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by dsnopek** [comment]
Ah, ok, sorry for the noise then! I just didn't RTFM closely enough. I just assumed that I had the correct permissions because node works for other stuff. I'll follow those instructions and give it another shot. Thanks!

**Comment by bcauldwell** [comment]
David, did installing globally work for you? Is there anything else I can
do to help you get this up and running?

On Thu, Jul 23, 2015 at 8:18 AM, David Snopek <notifications@github.com>
wrote:

> Closed #328 <https://github.com/kalabox/kalabox/issues/328>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/328#event-363512459>.
>



-- 
- Benjamin Cauldwell

**Comment by pirog** [comment]
@bcauldwell go over to #327
